portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -57,4 +60,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(comments));
   }
+
+  /**
+   * Convert a string into a integer. If not a natural number,
+   * return 0 with an error message.
```

Comment:
It's usually good practice to return "-1" in the case of an error. You can handle this case separately inside `ListCommentsServlet` 

Another point that comes to mind, which maybe you already thought through... if the user doesn't input a valid integer, should we show 0 comments or all of the comments? I'll leave it to you to decide which you would prefer :)

Reply:
Fixed! I decided to do all comments since the user would likely not want to see 0 comments